src/zenml/zen_stores/schemas/pipeline_run_schemas.py

```diff
@@ -269,6 +269,13 @@ def fetch_metadata_collection(self) -> Dict[str, List[RunMetadataEntry]]:
             for k, v in step_metadata.items():
                 metadata_collection[f"{s.name}::{k}"] = v
 
+        # Fetch the metadata related to the schedule of this run
+        if self.deployment is not None:
+            if schedule := self.deployment.schedule:
+                schedule_metadata = schedule.fetch_metadata_collection()
+                for k, v in schedule_metadata.items():
+                    metadata_collection[f"schedule:{k}"] = v
```

Comment:
What if I call a step `schedule`? 🤔

Reply:
There is one little difference. If there were a step called `schedule`, the corresponding metadata entries in this model would start with `schedule::` not `schedule:`.